### PR TITLE
feat: add unit tests for authentication middelware

### PR DESCRIPTION
**Pull request by MayankBansal12**
Date: 07-11-2024

Developer Name: Mayank Bansal

---

## Issue Ticket Number

#2217 
## Description
- Added unit test cases for authentication middleware inside `authenticate.test.js`
- Due to a missing import statement was getting an error while running tests. Fixed the error by adding the missing config statement in three files.

### Documentation Updated?

- [ ] Yes
- [x] No

### Under Feature Flag

- [ ] Yes
- [x] No

### Database Changes

- [ ] Yes
- [x] No

### Breaking Changes

- [ ] Yes
- [x] No

### Development Tested?

- [x] Yes
- [ ] No

## Test Coverage
<details>
<summary>Screenshot 1</summary>

added test cases

</details>

<details>
<summary>Screenshot 2</summary>

test coverage

</details>

### CODE REVIEW

File: models/userStatus.js [hunk 1]
```diff
@@ -23,6 +23,7 @@ const { userState } = require("../constants/userStatus");
 const discordRoleModel = firestore.collection("discord-roles");
 const memberRoleModel = firestore.collection("member-group-roles");
 const usersCollection = firestore.collection("users");
+const config = require("config");
```
Comment (pankajjs) [inline models/userStatus.js:26]: why this change?

File: utils/discord-actions.js [hunk 2]
```diff
@@ -1,4 +1,5 @@
 const jwt = require("jsonwebtoken");
+const config = require("config");
```
Comment (pankajjs) [inline utils/discord-actions.js:2]: same?

File: models/userStatus.js (hunk 1 above)
Comment (MayankBansal12) [reply models/userStatus.js:26]: While running the test, I got an error `config is not defined`. `config.get` is used in `userStatus.js` without importing the config file and hence the error. You can check the same here: https://github.com/Real-Dev-Squad/website-backend/blob/develop/models/userStatus.js#L26

File: services/discordMembersService.js [hunk 3]
```diff
@@ -1,5 +1,5 @@
 const jwt = require("jsonwebtoken");
-
+const config = require("config");
```
[the remark this answers is not in the document]
Comment (MayankBansal12) [reply services/discordMembersService.js:2]: Again, `config.get` was used and got the same error.
https://github.com/Real-Dev-Squad/website-backend/blob/develop/services/discordMembersService.js#L3

File: utils/discord-actions.js (hunk 2 above)
Comment (MayankBansal12) [reply utils/discord-actions.js:2]: Same error for `discord-actions.js`
https://github.com/Real-Dev-Squad/website-backend/blob/develop/utils/discord-actions.js#L3

File: models/userStatus.js (hunk 1 above)
Comment (pankajjs) [reply models/userStatus.js:26]: config is a global variable. How it it giving error in this file.
I am sure that config is being used in multiple places.

Comment (MayankBansal12) [reply models/userStatus.js:26]: I am not sure why it is giving error for this file. I only made this change to get rid of the errors that I encountered while running tests.
Let me know about the next steps.

Comment (MayankBansal12) [reply models/userStatus.js:26]: @pankajjs Can I resolve the conversation?

**Comment by pankajjs** [comment]
assert more expectation
